packages/editor-ui/src/components/InputPanel.vue

```diff
@@ -88,6 +88,27 @@ const {
 	focusedMappableInput,
 	isMappingOnboarded: isUserOnboarded,
 } = storeToRefs(ndvStore);
+
+const rootNode = computed(() => {
+	if (!activeNode.value) return null;
+
+	return props.workflow.getChildNodes(activeNode.value.name, 'ALL').at(0) ?? null;
+});
+
+/**
+ * True if active node itself is the root node or its root node has run
+ */
+const isRootNodeOrHasRootNodeRun = computed((): boolean => {
```

Comment:
This approach seems to be a little buggy. I have document a couple of bugs in Linear.
https://linear.app/n8n/issue/SUG-3/sub-node-ndvs-should-open-with-the-input-pane-toggle-on-mapping-when

Maybe here we can approach this differently. 
If node has "debugging" data, then show debugging mode. Otherwise, default to mapping mode.

Reply:
Thanks for the catch and suggestion. I agree checking whether the debugging data exists makes more consistency.

To implement this approach, I updated the logic to check the run data of the current node itself, which is the condition used in `RunData.vue` to detect the data. As far as I tested locally, issues raised in Linear are resolved by this change.

Reply:
Testing it now. This case now when parent has run but tool was not called does not make sense anymore. Because text says "Parent node hasn't run yet". 🤔 We might need to circle back to David with this. 

<img width="653" alt="Screenshot 2025-01-21 at 10 36 52" src="https://github.com/user-attachments/assets/f568d1c3-086d-466a-a7d6-7ee052591445" />

Reply:
Left a comment here https://linear.app/n8n/issue/SUG-3/sub-node-ndvs-should-open-with-the-input-pane-toggle-on-mapping-when#comment-ec036700

Reply:
Thanks for the catch once again! I've updated the condition according to what David suggested on Linear. [2027c49](https://github.com/n8n-io/n8n/pull/12642/commits/2027c4953ccbc1bb63cb3acbea6af262f5a0df1b)